### PR TITLE
Cw721 token indexes refactor

### PR DESCRIPTION
**Pull request by maurolacy**
An addendum to #327 / #340, improving the cw721 tokens by owner index, and updating the docs.

### CODE REVIEW

File: contracts/cw721-base/src/state.rs [hunk 1]
```diff
@@ -65,11 +65,7 @@ impl<'a> IndexList<TokenInfo> for TokenIndexes<'a> {
 
 pub fn tokens<'a>() -> IndexedMap<'a, &'a str, TokenInfo, TokenIndexes<'a>> {
     let indexes = TokenIndexes {
-        owner: MultiIndex::new(
-            |d, k| (Vec::from(d.owner.as_ref()), k),
-            "tokens",
-            "tokens__owner",
-        ),
+        owner: MultiIndex::new(|d, k| (d.owner.clone(), k), "tokens", "tokens__owner"),
```
Comment (ethanfrey) [inline contracts/cw721-base/src/state.rs:68]: Do you need clone here? Isn't it `|d: TokenInfo, k: Addr|`? Or is it `|d: &TokenInfo, k: Addr|`?

Inlining the proper types would definitely make this more readable if we use it as an learning example

**Review by ethanfrey — APPROVED** [review]
In general looks good. And by using `Addr` rather than `&Addr`, you avoid the lifetime issue.

Reviewing it, I realized some of the implied types are confusing and would be great to make explicit. Otherwise, good to merge

File: contracts/cw721-base/src/state.rs (hunk 1 above)
Comment (maurolacy) [reply contracts/cw721-base/src/state.rs:68]: It's the second one. That's why `clone` is needed.

Will do the type inlining.

**Review by ethanfrey — APPROVED** [review]
Good stuff, makes it a lot clearer.

**Pull request merged**